pkg/serverless/trace/inferredspan/inferred_span.go

```diff
@@ -72,3 +83,66 @@ func FilterFunctionTags(input map[string]string) map[string]string {
 
 	return output
 }
+
+func CreateInferredSpan(event string, inferredSpan InferredSpan) {
+	// Parse the event into the EventKey struct
+	eventSource, attributes := ParseEventSource(event)
+	switch eventSource {
+	case "apigateway":
+		CreateInferredSpanFromAPIGatewayEvent(eventSource, attributes, inferredSpan)
+	case "http-api":
+		CreateInferredSpanFromAPIGatewayHTTPEvent(eventSource, attributes, inferredSpan)
+	case "websocket":
+		CreateInferredSpanFromAPIGatewayWebsocketEvent(eventSource, attributes, inferredSpan)
+	}
+}
+
+func CompleteInferredSpan(
+	processTrace func(p *api.Payload),
+	endTime time.Time,
+	isError bool,
+	requestId string,
+	inferredSpan InferredSpan) {
+
+	if inferredSpan.IsAsync {
+		inferredSpan.Span.Duration = inferredSpan.Span.Start
+	} else {
+		inferredSpan.Span.Duration = endTime.UnixNano() - inferredSpan.Span.Start
+	}
+	if isError {
+		inferredSpan.Span.Error = 1
+	}
+	traceChunk := &pb.TraceChunk{
+		Priority: int32(sampler.PriorityNone),
+		Origin:   "lambda",
+		Spans:    []*pb.Span{inferredSpan.Span},
+	}
+
+	tracerPayload := &pb.TracerPayload{
+		Chunks: []*pb.TraceChunk{traceChunk},
+	}
+
+	processTrace(&api.Payload{
+		Source:        info.NewReceiverStats().GetTagStats(info.Tags{}),
+		TracerPayload: tracerPayload,
+	})
+}
+
+// GenerateSpan declares and initializes a new inferred span
+// with the SpanID and TraceID
+func GenerateSpan() InferredSpan {
+	var inferredSpan InferredSpan
+	inferredSpan.Span = &pb.Span{}
+	inferredSpan.Span.SpanID = rand.Random.Uint64()
+	inferredSpan.Span.TraceID = rand.Random.Uint64()
+	return inferredSpan
+}
+
+// IsExpected checks if the envVars required for inferred spans are enabled
+func IsExpected() bool {
```

Comment:
I don't think this needs to be a function, it can be a constant/variable instead. Environment variables are set before the agent starts running and cannot change at any time after that. So we don't need to recompute/recheck them over and over again. We can check once and then store the value forever.

Reply:
Also I wouldn't call this `IsExpected`, maybe something a bit more specific like `InferredSpansEnabled`